### DISCUSSION TITLE
checkbox 不能默认选中

### DISCUSSION THREAD

**Discussion by lixiaofa** [youzan/vant#11131] [General]
### 重现链接

https://codesandbox.io/s/vant-3-issue-template-forked-o54p4e?file=/src/App.vue

### Vant 版本

3.24

### 描述一下你遇到的问题。

默认值无法选中

### 重现步骤

默认值无法选中

**Comment by wjw-gavin** [comment]
`checked`值为引用类型数组时，需要绑定值来自数据源本身，`checked`来自`list`
![image](https://user-images.githubusercontent.com/19986739/194791464-0d511312-0c59-4b32-9afe-cad88c85381e.png)

**Comment by lixiaofa** [comment]
@wjw-gavin 好的，谢谢， 如果有二级的话    ， checked值也是要需要绑定值来自数据源本身吗
![image](https://user-images.githubusercontent.com/23230215/194792609-7c60fbdc-e6c4-4db3-a87e-1a7cb80465b8.png)
我应该怎么办？

**Comment by wjw-gavin** [comment]
其实当你要绑定默认值时，应该是从数据源中去匹配对应的勾选项，而不是重新定义一个“看上去”跟他一样的数据。看上去一样，就好比
![image](https://user-images.githubusercontent.com/19986739/194806222-12814b6d-1bb4-4880-a73c-a5a7ad2350f4.png)

![image](https://user-images.githubusercontent.com/19986739/194805471-53a924ba-59c4-4593-aa0d-efc96c2530aa.png)

**Comment by lixiaofa** [comment]
@wjw-gavin 不， 我的意思 是， 
![image](https://user-images.githubusercontent.com/23230215/194810539-bc0611b2-2faa-4e06-b382-01514749dfcd.png)

**Comment by wjw-gavin** [comment]
如果你的子项也使用`CheckboxGroup`的话，`CheckboxGroup`有个`change`事件，你在每次勾选或者取消勾选时都会触发`change`事件，你可以判断`change`回调参数中的数据`length`和你当前项的所有子项`length`是否一样，如果一样就代表全选了，然后你取到当前项并通过设置`checked`值来回显当前是否被选中。

**Comment by lixiaofa** [comment]
@wjw-gavin   我已经试过了， 谢谢。 都用的是CheckboxGroup  ， 如果子项我只选了一个， 你的checked 数据结构是什么 ，

checked 和list(原始数据)是绑定的，      明显有bug ， 个人觉得， checked 不能用全等 ， 只要属性相同就可以了, 而且日常开发中，checked 对象数组的 场景比较多 ， 

大佬，有购物车的三级选中   ， 用vant checkbox 的实现的案例吗， 我看看大佬是怎么写的， 或者好的文章 ， 谢谢

**Comment by lixiaofa** [comment]
@wjw-gavin   大佬你能写一个购物车的三级选中吗，  用vant checkbox ， 让我学习学习

**Comment by lixiaofa** [comment]
@wjw-gavin  大佬 购物车三级选中功能建议怎么去实现， 绑定对象，还是id 属性

**Reply by wjw-gavin** [reply]
1、你的购物车是三级的？
2、建议绑定id;

我根据你提供的数据，写了一个二级的购物车
https://codesandbox.io/s/vant-3-issue-template-forked-hnns3y?file=/src/App.vue
三级的你可以参考自己写一下😂

**Reply by lixiaofa** [reply]
好的， 谢谢大佬

**Comment by chenjiahan** [comment]
I will convert issue to a discussion because is is not a bug or feature.
